File: articles/grit-in-positive-psychology/fortran/grit_positive_psychology_score.f90
program grit_positive_psychology_score
  implicit none
  real :: perseverance_effort, durable_interest, grit_score

  perseverance_effort = 0.78
  durable_interest = 0.66
  grit_score = 0.60 * perseverance_effort + 0.40 * durable_interest

  print *, "Synthetic grit score for positive psychology model:", grit_score
end program grit_positive_psychology_score

File: articles/grit-in-positive-psychology/c/grit_positive_psychology_score.c
#include <stdio.h>

double grit_score(double perseverance_effort, double durable_interest) {
    return 0.60 * perseverance_effort + 0.40 * durable_interest;
}

int main(void) {
    double perseverance_effort = 0.78;
    double durable_interest = 0.66;
    printf("Synthetic grit score for positive psychology model: %.3f\n",
           grit_score(perseverance_effort, durable_interest));
    return 0;
}
